src/module_wrap.cc

```diff
@@ -847,8 +847,11 @@ Maybe<URL> PackageExportsResolve(Environment* env,
         std::string target(*target_utf8, target_utf8.length());
         if (target.back() == '/' && target.substr(0, 2) == "./") {
           std::string subpath = pkg_subpath.substr(best_match_str.length());
+          URL url_prefix(target, pjson_url);
           URL target_url(target + subpath, pjson_url);
-          return FinalizeResolution(env, target_url, base);
+          if (target_url.path().find(url_prefix.path()) == 0) {
```

Comment:
@guybedford This aligns the behavior with what we in CJS - we don't allow `pkg/exposed-path/../foo` to break out of the mapping for `exposed-path`.

Reply:
Nice, seems sensible to me. Why doesn't the same apply to exact matches though?

Reply:
This only verifies that the importer of the package stays within the boundary, the package could map to something else explicitly if it wanted to. For exact matches the importer doesn't have the option to try to break out of the prefix.

Reply:
That distinction makes sense. Perhaps one of the validations of package targets should be that it stays within the package and resolves as a valid URL (and report errors on the failure of either of those). But certainly that could be implemented as a follow-up.

Reply:
I'm torn on that. I think there's value in letting packages specify external targets. E.g. "I'm just a polyfill for this `std:` module" would be an example. For `fs:`-only there's less of a real use case but I'm not sure it's worth special-casing.